src/components/Table.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import Loading from './Loading'
+
+const Table = ({headers, data, headerTitle, toggle, loading}) => {
+  return (
+    <table className="my-2 border-2 border-gray-100 ">
+      {/* table header */}
+        <thead className="">
+          <tr className="my-2">
+            {headerTitle.map((header) => (
+              <th key={header} className="px-3 border-2 capitalize">{header}</th>
+            ))}
+          </tr>
+        </thead>
+        {/* table body */}
+        <tbody className="">
+          {data != undefined && data.length > 0 ? (
+            data.map((item, index) => (
+              <tr
+                key={item.ticker}
+                className={`${index % 2 == 0 && toggle ? "bg-gray-100 text-black" : ""} 
+               h-12 border-gray-300 px-4 py-2 cursor-pointer hover:text-violet-500`}
+                onClick={() => handleClick(item.ticker)}
+              >
+                {headers.map((header) => (
+                  <td key={header} className="px-3 border-r-2 ">
+                    {item[header]}
+                  </td>
+                ))}
+              </tr>
+            ))
+          ) : (
+            <>
+            {loading ? <Loading/> : 
+            <tr className="py-2">
+              <td>"No data found"</td>
+            </tr>}
```

Comment:
this will only show loader when data is not there if data is there then it will not show.

better move `loading` outside of condition.

also if you are showing `No data` and `loader` here then no need to show  in Infinite scroll